src/database/migrations/20240715093559-add-file-type-entities.js

```diff
@@ -0,0 +1,253 @@
+/** @type {import('sequelize-cli').Migration} */
+
+module.exports = {
+	async up(queryInterface, Sequelize) {
+		const defaultOrgId = queryInterface.sequelize.options.defaultOrgId
+		if (!defaultOrgId) {
+			throw new Error('Default org ID is undefined. Please make sure it is set in sequelize options.')
+		}
+
+		const entitiesArray = [
+			{
+				entityType: 'file_types',
+				entities: [
+					{ value: 'images', label: 'Images' },
+					{ value: 'document', label: 'Document' },
+					{ value: 'videos', label: 'Videos' },
+					{ value: 'audio', label: 'Audio' },
+				],
+				has_entities: true,
+				validation: { required: false },
+				model: 'tasks',
+			},
+		]
+
+		const entityTypeFinalArray = entitiesArray.map((entity) => {
+			const { entityType, has_entities, validation, model } = entity
+			return {
+				value: entityType,
+				label: convertToWords(entityType),
+				data_type: 'ARRAY[STRING]',
+				status: 'ACTIVE',
+				updated_at: new Date(),
+				created_at: new Date(),
+				created_by: 0,
+				updated_by: 0,
+				allow_filtering: false,
+				organization_id: defaultOrgId,
+				has_entities,
+				allow_custom_entities: false,
+				validations: validation ? JSON.stringify(validation) : null,
+			}
+		})
+		await queryInterface.bulkInsert('entity_types', entityTypeFinalArray, {})
+
+		const entityTypes = await queryInterface.sequelize.query('SELECT * FROM entity_types', {
+			type: queryInterface.sequelize.QueryTypes.SELECT,
+		})
+
+		const entityModelMapping = entitiesArray.map((entity) => {
+			const entityType = entityTypes.find((type) => type.value === entity.entityType)
+			return {
+				entity_type_id: entityType.id,
+				model: entity.model,
+				status: 'ACTIVE',
+				updated_at: new Date(),
+				created_at: new Date(),
+			}
+		})
+
+		await queryInterface.bulkInsert('entities_model_mapping', entityModelMapping, {})
+
+		const entitiesFinalArray = entityTypes.reduce((acc, eachType) => {
+			const entityData = entitiesArray.find((entity) => entity.entityType === eachType.value)
+			if (entityData && eachType.has_entities) {
+				entityData.entities.forEach((eachEntity) => {
+					acc.push({
+						...eachEntity,
+						entity_type_id: eachType.id,
+						type: 'SYSTEM',
+						status: 'ACTIVE',
+						created_at: new Date(),
+						updated_at: new Date(),
+						created_by: 0,
+						updated_by: 0,
+					})
+				})
+			}
+			return acc
+		}, [])
+
+		await queryInterface.bulkInsert('entities', entitiesFinalArray, {})
+
+		//add model mapping for learning resource
+		const learningResourceEntityType = await queryInterface.sequelize.query(
```

Comment:
@adithyadinesh0412 Create different migration files for learning resources. One for file entity types one for learning resources. Both are different. Also migration file name just tells file type.
Add in comment on top of each migration file what it does and date author etc information.